baybe/surrogates/utils.py

```diff
@@ -57,156 +55,101 @@ def _prepare_targets(y: Tensor) -> Tensor:
     return y.to(DTypeFloatTorch)
 
 
-def catch_constant_targets(model_cls: type[Surrogate]):
-    """Wrap a ``Surrogate`` class that cannot handle constant training target values.
+def catch_constant_targets(cls: type[Surrogate], std_threshold: float = 1e-6):
+    """Make a ``Surrogate`` class robustly handle constant training targets.
+
+    More specifically, "constant training targets" can mean either of:
+        * The standard deviation of the training targets is below the given threshold.
+        * There is only one target and the standard deviation cannot even be computed.
 
-    In the wrapped class, these cases are handled by a separate model type.
+    The modified class handles the above cases separately from "regular operation"
+    by resorting to a :class:`baybe.surrogates.naive.MeanPredictionSurrogate`.
 
     Args:
-        model_cls: A ``Surrogate`` class that should be wrapped.
+        cls: The :class:`baybe.surrogates.base.Surrogate` to be augmented.
+        std_threshold: The standard deviation threshold below which operation is
+            switched to the alternative model.
 
     Returns:
-        A wrapped version of the class.
+        The modified class.
     """
+    from baybe.surrogates.naive import MeanPredictionSurrogate
```

Comment:
Have moved it here because a top-level import is not possible due to import cycles. Having it here saves one additional import because if I move it to `_posterior_new`, I also need to copy it to `_fit_new`. So think I can resolve this, right? Please unresolve if unhappy